### PR TITLE
How to Optimize WordPress Website Speed

### PR DESCRIPTION
**Pull request by Tsanguu**
Added index.md, hero.jpg, and 4 article images.
The article is about WordPress speed optimization.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by hectorkambow** [comment]
Plagiarism check done

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+Cache](https://wordpress.org/plugins/wp-fastest-cache/), and [WP Super
+Cache](https://wordpress.org/plugins/wp-super-cache/).
+
+### Optimize Images
+
+An image is worth a thousand words. You cannot avoid using images on
+your website. Images increase engagement and bring life to the web
+content. In their original formats, image files are huge and could be
+more hurting than helping. The most common image formats used for the
+web are PNG and JPEG. The PNG image format is uncompressed while JPEG is
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:102]: You could mention the WebP image format.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+[Optimole](https://wordpress.org/plugins/optimole-wp/), and [WP
+Smush](https://wordpress.org/plugins/wp-smushit/). Keep your image under
+100 KB for the best performance.
+
+### Delete unused themes, plugins, and media files
+
+This is a crucial website maintenance task. You should get rid of themes
+and plugins that are not in use. The plugins and themes take space in
+the server causing unnecessary strain. This involves deleting all the
+installed themes and deactivated plugins.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:124]: **redundant** themes not every installed one. Beginners may get confused.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+define('WP_POST_REVISIONS', 4 )
+```
+
+![WordPress wp-config editing](wp-config.php-edit.jpg)
+
+### Update PHP to the latest version
+
+PHP is opensource and undergoing active development to improve on
+performance, fix bugs, enhance security, and other features. Therefore,
+there is no doubt that the latest PHP version is more enhanced in terms
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:192]: **open source**

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+of performance compared to the previous versions. Keep your PHP version
+updated to get the benefits. PHP update can be as simple as selecting
+the latest version for cPanel users. Check with your hosting provider on
+how to update the PHP version.
+
+### Use Content Delivery Network (CDN)
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:198]: You could expand here on the current state of PHP versions in WordPress. 

What's the minimum supported version and the max? Which versions are in development and which are common hosting providers using? When does support end for different PHP versions?

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
Great article, @Tsanguu. I'm sure beginners to WordPress optimisation would find it helpful.

I've suggested areas you could expand upon to add value and revisions you could make to aid readability.

I've noticed there were a few cases of inconsistent capitalisation and using the singular form instead of plural. Please fix these and take them on board for future articles.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
+The ultimate measure of website performance is loading speed, the amount
+of time it takes to load a web page. Web users are relying more on
+mobile to access web content and services. If you run any website, this
+is very evident in the data you have. Users are also demanding better
+web experience thus the need to improve your WordPress website
+performance. To improve overall website performance, several measures
+need to be taken. This article is a comprehensive guide on the best ways
+to speed up WordPress website.
+
+## Why website speed matter?
```
[the remark this answers is not in the document]
Comment (Tsanguu) [reply articles/how-to-optimize-wordpress-website-speed/index.md]: Noted and corrected.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
+The ultimate measure of website performance is loading speed, the amount
+of time it takes to load a web page. Web users are relying more on
+mobile to access web content and services. If you run any website, this
```
[the remark this answers is not in the document]
Comment (Tsanguu) [reply articles/how-to-optimize-wordpress-website-speed/index.md]: Noted and corrected.

**Comment by Tsanguu** [comment]
@louisefindlay23 I have worked on the revisions.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
+The ultimate measure of websites performance is loading speed. This is the amount
+of time it takes to load a web page. Web users are relying more on
+mobiles to access web content and services and thus more demand for optimized websites. If you run any website, this
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:3]: manage any website**s** would flow better. Sorry, I should have been clearer. This occurrence of website should be plural not the first one.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
+The ultimate measure of websites performance is loading speed. This is the amount
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:1]: Sorry, I should have been clearer. This occurrence of website should be singular, the second one should be plural.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+of time it takes to load a web page. Web users are relying more on
+mobiles to access web content and services and thus more demand for optimized websites. If you run any website, this
+is very evident in your analytical data. Users are also demanding better web experience thus the need to improve your WordPress website performance. To improve overall website performance, several measures
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:4]: demanding **a** better. Thus **there is a** need would flow better.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+need to be taken. This article is a comprehensive guide on the best ways to speed up WordPress website.
+
+## Why Website Speed Matters?
+
+If your website takes longer than 2 seconds to load, the bounce rate increases.
+This means you will have reduced conversions and a decrease in
+revenue. According to the [Google SEO guide](https://www.google.com/search/howsearchworks/algorithms/), "page speed" of sites is among the parameters considered in the search engine ranking algorithm since January 2018. Slower websites are ranked lower as compared to faster loading websites. Fast loading web pages lead to improved user experience, rise in page views, and contribute to higher search engine rankings.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:11]: Slower websites are ranked lower than faster websites would be simpler.

Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:11]: webpages should be one word not two separate words. Consider omitting loading in fast loading webpages to simplify the sentence.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+## Tools for Testing Website Page Speed
+
+WordPress speed is not dependent on whether it feels slow on the developer's computer. It might load faster as a result of browser caching
+yet it is very slow for a first-time visitor. Speed testing tools come
+in hand in determining your page speed. It is also worth noting that speed
+may vary with geographical locations. Therefore, you need to test speed from different locations. This article will go on to talk about this on the [CDN section](#use-content-delivery-network-(CDN)). The faster the website the better.
+Even improving your website speed by milliseconds matters. A load time
+under 2 seconds is recommended.
+
+There are several tools for testing website speed. Some of them are free
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:22]: are free **and**

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+others are paid. Test your website speed before optimization and after
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:23]: before and after optimisation would flow better.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+Shared hosting, VPS hosting, and managed WordPress hosting. Shared
+hosting is the most popular. Companies like Namecheap, Bluehost,
+HostGator, SiteGround, and GoDaddy offer shared hosting. Your website is
+likely to be hosted with more than 200 other websites on the same
+server. When the other sites hosted together with yours get an upsurge in traffic, there is a higher likelihood of the server performance being
+impacted thus making your website slow.
+
+VPS (Virtual Private Server) hosting requires you to be able to do server
+configuration and server management. VPS costs from $5 a month. Some of
+the VPS providers are Amazon, Linode, and Digital Ocean. Amazon, Linode, and Digital Ocean be more complex to configure for beginners. Providers such as [Runcloud](https://runcloud.io/), [SpinupWP](https://spinupwp.com/), and [GridPane](https://gridpane.com/) provide VPS that are easier to manage and made for WordPress. The VPS approach cuts on cost. However, you will be responsible for server optimization and if something breaks.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:59]: provide VPS should be plural.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+This list is not ordered by any criteria. All are good practices for
+website optimization. Consider them for your WordPress website optimization.
+
+### Choose a Good Hosting Platform
+
+Web hosting services can be sorted into three categories;
+
+Shared hosting, VPS hosting, and managed WordPress hosting. Shared
+hosting is the most popular. Companies like Namecheap, Bluehost,
+HostGator, SiteGround, and GoDaddy offer shared hosting. Your website is
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:52]: Consider linking to all the providers you mention.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+This is a crucial website maintenance task. You should get rid of themes
+and plugins that are not in use. The plugins and themes take space in
+the server causing unnecessary strain. This involves deleting installed themes that are not in use and deleting deactivated plugins.
+
+### Use Well-optimized Themes
+
+Don't select some beautiful and impressive looking WordPress themes at
+the expense of performance. Pay special attention to theme optimization. Your
+WordPress theme should be elegant, lightweight, SEO friendly, fully
+responsive, and mobile optimized.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:123]: Perhaps you could suggest some WP themes that are well-optimised and responsive like Astra and Page Builder Framework.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+
+### Use Lazy Loading for Images and Videos
+
+Lazy loading is a web optimization technique whereby, visible content is
+loaded first while contents appearing down the fold download and
+rendering is delayed. Instead of loading all images and videos, Lazy
+loading downloads and render only what is visible to the user's screen.
+As the user scrolls down the page, more content is loaded. This
+significantly increases the loading speed of the content that is visible
+to the user.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:149]: You could mention how to activate lazy-loading images such as the built-in functionality in the latest WP update.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+update a page or a post, another version of the post is created and
+saved in the database. Overtime, post revisions can increase to a point
+whereby they slow your WordPress database. In the screenshot below, post
+revisions have accumulated to 93.
+
+![WordPress post revisions screenshot](wordpress-post-revisions-screenshot.jpeg)
+
+The solution for post revisions is to clean them from the database. You
+can use the WP-Optimize plugin to clean post revisions as shown in the
+screenshot below.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:163]: Perhaps provide an explicit warning to the user that only four post revision will be saved so they will only be able to access the last four versions of their page or post.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+
+PHP is open source and undergoing active development to improve on
+performance, fix bugs, enhance security, and other features. Therefore,
+there is no doubt that the latest PHP version is more enhanced in terms
+of performance compared to the previous versions. Keep your PHP version
+updated to get the benefits. Currently, WordPress works with PHP 5.6.20 or greater but the recommended version is 7.4 or greater. Refer [here](https://wordpress.org/about/requirements/). Most hosting providers have PHP version 7+. PHP update can be as simple as selecting the latest version for cPanel users. Check with your hosting provider on how to update PHP version. According to the  PHP [official website](https://www.php.net/supported-versions.php), the latest version is 7.4, released Nov 2019. Each version of PHP is fully supported for two years from its initial stable release to fix bugs and security issues. Check when your PHP version support will end on PHP [official website](https://www.php.net/supported-versions.php).
+
+### Use Content Delivery Network (CDN)
+
+CDN is a network of servers linked together in different locations
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:190]: **A** CDN.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 19]
```diff
@@ -0,0 +1,214 @@
...
+users located in Germany. Some hosting companies such as
+[Amazon](https://aws.amazon.com/) and [Linode](http://linode.com/) have
+servers in different geographical locations. You are advised to select a
+server that is closest to the majority of your users. Among the market leaders in CDN services
+are Cloudflare, Akamai, Amazon CloudFront, IBM Cloud, Internet Services,
+Azure CDN, Google Cloud CDN, and Fastly.
+
+### Conclusion
+
+Fast website boosts your search engine rankings, improves search
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:210]: website**s**

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
Thanks for making the requested changes, @Tsanguu and taking on my feedback on areas to expand. I'm glad I'm able to add value to your article.

For future articles, consider using a free tool like [Grammarly](https://www.grammarly.com) which will help in both spelling wnd grammar.

There are a few more changes you need to make and some suggestions additions.

**Comment by Tsanguu** [comment]
@louisefindlay23 I have worked on the suggestions. This time I have used Grammarly.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+
+```php
+define('WP_POST_REVISIONS', 4 )
+```
+
+![WordPress wp-config editing](wp-config.php-edit.jpg)
+
+### Update PHP to the Latest Version
+
+PHP is open-source and undergoing active development to improve on performance, fix bugs, enhance security, and other features. Therefore, there is no doubt that the latest PHP version is more enhanced in terms of performance compared to the previous versions. Keep your PHP version updated to get the benefits.
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:138]: Consider giving a brief explanation of what PHP is. Readers new to WordPress may not understand that PHP is a programming language. Also, consider giving a tangible metric on how much better new version of PHP are. For example, the speed increase in moving from PHP 5.6 to PHP 7.

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
@Tsanguu, thanks for making the changes and using Grammarly. The article is much improved. I've made some minor edits for readability and have some suggestions about how you could improve the PHP section.

**Comment by Tsanguu** [comment]
@louisefindlay23 Thank you for the insight. I have worked on the suggestion.

**Comment by louisefindlay23** [comment]
Great work, @Tsanguu. It's ready for tech review.

File: articles/how-to-optimize-wordpress-website-speed/index.md [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+See above screenshot of WP-Optimize 3.1.2 free version database optimization
+
+### Use Lazy Loading for Images and Videos
+
+Lazy loading is a web optimization technique where visible content is loaded first while content appearing below the fold is only downloaded and rendering when necessary. Instead of loading every image and video on a webpage, lazy loading downloads and render only what is visible to the user's screen.
+
+As the user scrolls down the page, more content is loaded. This significantly increases the loading speed of the content that is visible
+to the user. 
+
+You can activate lazy-loading using plugins such as [Lazy Load for Images](https://wordpress.org/plugins/lazy-load-for-images/) and [BJ Lazy Load](https://wordpress.org/plugins/bj-lazy-load/).
```
Comment (louisefindlay23) [inline articles/how-to-optimize-wordpress-website-speed/index.md:112]: The BJ Lazy Load plugin hasn't been updated in 3 years or been tested in the last three major versions of WordPress so I would recommend removing it. Perhaps highlight the native lazy loading WordPress feature in WP 5.5? https://kinsta.com/blog/wordpress-5-5/#native-image-lazyloading-in-wordpress-core

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
Looks great, @Tsanguu. One of the plugins you mentioned hasn't been updated in a couple of years so I'd recommend replacing it. 

Once you've fixed that, I'm happy to pass it on to the final review to be published.

**Comment by louisefindlay23** [comment]
Hey, @Tsanguu. Just noticed this is your first article. Congratulations. Can you add a hero image and bio please so we can add it to your article?

**Comment by Tsanguu** [comment]
@louisefindlay23  Thank you. See the attached files.
[index.txt](https://github.com/section-io/engineering-education/files/5189082/index.txt)
avatar
hero

**Comment by louisefindlay23** [comment]
@hectorkambow, can you create an author page with the bio and headshot provided?

File: articles/how-to-optimize-wordpress-website-speed/index.md (hunk 21 above)
Comment (Tsanguu) [reply articles/how-to-optimize-wordpress-website-speed/index.md]: @louisefindlay23 This was a very helpful insight. I worked on it.

**Comment by hectorkambow** [comment]
@Tsanguu 👍 
? Did you happen to make that hero image yourself?

**Comment by Tsanguu** [comment]
I downloaded the main image on https://unsplash.com/photos/jBAyrfdYZEc then made some modification.

**Comment by hectorkambow** [comment]
> @hectorkambow I downloaded the main image on https://unsplash.com/photos/jBAyrfdYZEc then made some modifications.

very cool 👍  love it! Great work

**Comment by hectorkambow** [comment]
The article has been merged and should be live in a moment. @Tsanguu

**Comment by Tsanguu** [comment]
@hectorkambow and reviewer @louisefindlay23 thank you. Looking forward to writing more.

**Pull request merged**